pkg/registry/generic/storage_factory.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package generic
+
+import (
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/storage"
+)
+
+// StorageFactory is a function signature for producing
+// a storage.Interface from given parameters.
+type StorageFactory func(
+	storageInterface storage.Interface,
+	capacity int,
+	objectType runtime.Object,
+	resourcePrefix string,
+	namespaceScoped bool,
+	newListFunc func() runtime.Object) storage.Interface
+
+// Returnss given 'storageInterface' without any decoration.
+func ForwardStorage(
```

Comment:
PassthroughStorage? DefaultFactory? I don't care for "Forward" but I don't really have better suggestions. NoDecoration isn't so bad either ;)

Oh, how about "UndecoratedStorage"?